### ISSUE TITLE
Doesn't work in 0.217.25

### ISSUE THREAD

**Issue by SanyaBane** [yoship1639/ValheimVRM#41]
Mod does not work in 0.217.25 (did not test with previous versions, since have no idea how to do it).
Game itself with BePis works fine.
It also works if I only copy content of "plugin" folder into BePis "plugins".
But if I copy dll files from "DataManaged" to "Managed" (with 5 files being replaced in process), then after launching game it will only show 2 splash screens, and then immediately crash, right before game's "Main Menu" should appear.

**Comment by nightshade231** [comment]
I've been getting this same issue since about 3AM this morning, both with 2202 and 2201 builds of BepInEX. Also with 0.217.25 (n-15) as freshly installed.

It would throw "CamZoomIn"/"CamZoomOut" bind warnings (yellow) in console before throwing "GetPixel called on an undefined mip (valid values 0 - 6" as a fatal error. Reverting the data/managed back to pre-modified files lets the game launch, but the mod not loaded.

**Comment by nightshade231** [comment]
> I've been getting this same issue since about 3AM this morning, both with 2202 and 2201 builds of BepInEX. Also with 0.217.25 (n-15) as freshly installed.
> 
> It would throw "CamZoomIn"/"CamZoomOut" bind warnings (yellow) in console before throwing "GetPixel called on an undefined mip (valid values 0 - 6" as a fatal error. Reverting the data/managed back to pre-modified files lets the game launch, but the mod not loaded.

Further one-by-one testing results in the file "UnityEngine.CoreModule.dll" tripping the error, all other files do not cause the error.

**Comment by Rahat47313** [comment]
yea, I have the same issue, can't launch game if this is installed...
Also, noticed that the last update of this mod is jul 22, 2022...has this amazing mod been abandoned?

**Comment by RemainIndoors1** [comment]
> yea, I have the same issue, can't launch game if this is installed... Also, noticed that the last update of this mod is jul 22, 2022...has this amazing mod been abandoned?

my friend helped me figure out there is a way to make this work with the new update. You need to download and add the latest BepInEx as they released an update around the same time the new patch came out. After that, copy the files from the BepInEx_plugins folder as you normally would, and add your ValheimVRM folder the same as you normally would.

However, when you copy the files from valheim_Data_Managed to the valheim_Data - Managed folder, you need to only copy the following files:

- FastSpringBone.dll
- FastSpringBone10.dll
- MToon.dll
- System.ValueTuple.dll
- UniGLTF.dll
- UniGLTF.Samples.ScreenSpace.dll
- UniHumanoid.dll
- VRM.dll
- VRM10.dll
- VrmLib.dll
- VRMShaders.GLTF.IO.Runtime.dll
- VRMShaders.GLTF.UniUnlit.Runtime.dll
- VRMShaders.VRM.IO.Runtime.dll
- VRMShaders.VRM10.Format.Runtime.dll
- VRMShaders.VRM10.MToon10.Runtime.dll

basically, if you overwrite any of the files in that folder, it will cause the game to crash when you try to start it, but adding just the files that aren't there already seems to work so far.

Hope that helps :)

**Comment by SanyaBane** [comment]
> However, when you copy the files from valheim_Data_Managed to the valheim_Data - Managed folder, you need to only copy the following files:

Wow, indeed it works. I can't understand why other dlls (which replace the ones which game has) are in mod directory then. 
Was it just a mod developer mistake and those dlls can be safely skipped, or not replacing them might cause some bugs later?

Anyway, it works and I'm really glad for it. Thank you for sharing information.

**Comment by RemainIndoors1** [comment]
> > However, when you copy the files from valheim_Data_Managed to the valheim_Data - Managed folder, you need to only copy the following files:
> 
> Wow, indeed it works. I can't understand why other dlls (which replace the ones which game has) are in mod directory then. Was it just a mod developer mistake and those dlls can be safely skipped, or not replacing them might cause some bugs later?
> 
> Anyway, it works and I'm really glad for it. Thank you for sharing information.

no problem. it did work previously overwriting the files, but I'm guessing some of the files being overwritten were modified or newer versions of Unity dlls that the updates to the game now depend on. I doubt the mod developers modified the Unity dlls, so those may have just been newer versions that the game wasn't using yet.  There is the potential for some bugs though, so it's hard to say if something won't pop up with shaders or rendering issues or something without a lot of testing.

I'm surprised it works without updating the existing dlls, but glad to see it does so far, because I'm not crazy about the default character models looking like potatoes :-/

**Comment by Rahat47313** [comment]
> > yea, I have the same issue, can't launch game if this is installed... Also, noticed that the last update of this mod is jul 22, 2022...has this amazing mod been abandoned?
> 
> my friend helped me figure out there is a way to make this work with the new update. You need to download and add the latest BepInEx as they released an update around the same time the new patch came out. After that, copy the files from the BepInEx_plugins folder as you normally would, and add your ValheimVRM folder the same as you normally would.
> 
> However, when you copy the files from valheim_Data_Managed to the valheim_Data - Managed folder, you need to only copy the following files:
> 
> * FastSpringBone.dll
> * FastSpringBone10.dll
> * MToon.dll
> * System.ValueTuple.dll
> * UniGLTF.dll
> * UniGLTF.Samples.ScreenSpace.dll
> * UniHumanoid.dll
> * VRM.dll
> * VRM10.dll
> * VrmLib.dll
> * VRMShaders.GLTF.IO.Runtime.dll
> * VRMShaders.GLTF.UniUnlit.Runtime.dll
> * VRMShaders.VRM.IO.Runtime.dll
> * VRMShaders.VRM10.Format.Runtime.dll
> * VRMShaders.VRM10.MToon10.Runtime.dll
> 
> basically, if you overwrite all of the files in that folder that already exist, it will cause the game to crash when you try to start it, but adding just the files that aren't there already seems to work so far.
> 
> Hope that helps :)
> 
> ** edit ** : if anybody's having problems getting it to work, you might need to uninstall the game and delete the Valheim folder from the Steam - steamapps - common folder, then reinstall and start over.

Thanks a lot!!! its working again, I can use the 2B mod again, thanks to you
